examples/get_matching/server.c

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (C) Bull S.A.S - 2024
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program; if
+ * not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ *
+ * BXI Low Level Team
+ *
+ */
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "portals4.h"
+#include "portals4_ext.h"
+
+#include "get_me.h"
+
+/*
+ * this file represent a server preparing messages which can be get from a client
+ *
+ * Usage: ./server [max messages possible to get]
+ *
+ * The server will exit after the specified number of messages have been got.
+ * By default, the server wait forever.
+ */
+
+/* used to perform a MEAppend*/
+void add_me(ptl_handle_ni_t nih, ptl_index_t pti, ptl_me_t me, ptl_handle_me_t meh,

```

Comment:
Why do we need two return codes? Can't we only have a single return code, that cleans up when needed?

Meaning that the function returns an `int`, and if the PtlEQWait fails then we perform a PtlLEUnlink? Thoug the EQWait should never fail, we could abort instead.

Also the argument needs to be a `ptl_handle_me_t *meh`, else we don't save it's value